### PR TITLE
Update a11y-use-next-tooltip to be a11y-use-accessible-tooltip and make the changes accordingly

### PR DESCRIPTION
**Pull request by broccolinisoup**
Since Tooltip v2 is now exported from `@primer/react` bundle and Tooltip v1 from `@primer/react/deprecated`, this PR updates the code accordingly and renames the rule.

### CODE REVIEW

File: src/rules/a11y-use-accessible-tooltip.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            if (!hasOtherImports) {
+              // If Tooltip is the only import and no existing @primer/react import, replace the whole import statement
+              if (!hasRootImport) fixes.push(fixer.replaceText(node.source, `'@primer/react'`))
+              if (hasRootImport) {
+                //  remove the entire import statement
+                fixes.push(fixer.remove(node))
+                // find the last specifier in the existing @primer/react import and insert Tooltip after that
+                const rootImport = sourceCode.ast.body.find(statement => {
+                  return statement.type === 'ImportDeclaration' && statement.source.value === '@primer/react'
+                })
```
Comment (francinelucca) [inline src/rules/a11y-use-accessible-tooltip.js:67]: Wondering if there's a reason we can't reuse the `rootImport` from line 43 here?

**Review by francinelucca — COMMENTED** [review]
Looks good! Just one question: is there any potential side-effects to changing the rule name? (e.g., if it's being overridden in current code usage would a bunch of comments need to be updated or something like that?)

File: src/rules/a11y-use-accessible-tooltip.js (hunk 1 above)
Comment (broccolinisoup) [reply src/rules/a11y-use-accessible-tooltip.js]: We can re-use! I missed that, due to using the wrong method (filter vs find.) I updated the code 🤗 Thanks for catching that.

**Comment by broccolinisoup** [comment]
> Looks good! Just one question: is there any potential side-effects to changing the rule name? (e.g., if it's being overridden in current code usage would a bunch of comments need to be updated or something like that?)

The rule is currently off at dotcom - we haven't started using it yet so thankfully no need to update a bunch of line etc. However [I caught 2 usages in the helphub](https://github.com/primer/eslint-plugin-primer-react/pull/270#pullrequestreview-2418751349) so these need to be updated. I'll mark this as major so that we can communicate this change.

File: .changeset/gorgeous-frogs-cover.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"eslint-plugin-primer-react": patch
```
Comment (broccolinisoup) [inline .changeset/gorgeous-frogs-cover.md:2]:
```suggestion
"eslint-plugin-primer-react": major
```

**Review by francinelucca — APPROVED** [review]
🚀🚀

**Review by joshblack — APPROVED** [review]

**Pull request merged**